spec/jobs/run_submissions_job_spec.rb

```diff
@@ -43,15 +42,34 @@
   end
 
   describe "queuing a submission" do
+    before do
+      perform_enqueued_jobs { described_class.perform_later(good_submission.id) }
+    end
+
     it "runs the submission on the tests" do
-      perform_enqueued_jobs { job }
       expect(Execution.where(submission_id: good_submission.id).pluck(:output)).to match_array(
         Test.find([test_consec_3.id, test_consec_5.id]).pluck(:expected_output))
     end
 
     it "does not create duplicate Execution records" do
-      perform_enqueued_jobs { job }
-      expect{ perform_enqueued_jobs { job } }.to_not change{ Execution.count }
+      expect {
+        perform_enqueued_jobs { described_class.perform_later(good_submission.id) }
+      }.to_not change{ Execution.count }
+    end
+
+    it "does not rerun on old tests" do
+      expect(TestExecutor).to_not receive(:run_test)
+      perform_enqueued_jobs { described_class.perform_later(good_submission.id) }
+    end
+
+    let(:consec_6) { create(:consec, n: 6) }
```

Comment:
It's just that these declarations don't really make sense in the context of the other tests.